pkg/controller/masteruserrecord/sync.go

```diff
@@ -45,20 +45,39 @@ type Synchronizer struct {
 }
 
 func (s *Synchronizer) synchronizeSpec() error {
-	if !reflect.DeepEqual(s.memberUserAcc.Spec, s.recordSpecUserAcc.Spec) {
-		// when UserAccount spec in record is updated - is not same as in member
-		s.memberUserAcc.Spec = s.recordSpecUserAcc.Spec
+
+	if !s.isSynchronized() {
 		if err := updateStatusConditions(s.hostClient, s.record, toBeNotReady(updatingReason, "")); err != nil {
 			return err
 		}
+		s.memberUserAcc.Spec.UserAccountSpecBase = s.recordSpecUserAcc.Spec.UserAccountSpecBase
```

Comment:
could you just do `s.memberUserAcc.Spec = s.recordSpecUserAcc.Spec`?

Reply:
@xcoulon the specs are two different types now, so we need to compare the `UserAccountSpecBase` i think, unless i missed something?